pinot-query-runtime/src/main/java/org/apache/pinot/query/service/QueryConfig.java

```diff
@@ -59,6 +62,12 @@ public class QueryConfig {
   public static final String KEY_OF_SERVER_RESPONSE_STATUS_ERROR = "ERROR";
   public static final String KEY_OF_SERVER_RESPONSE_STATUS_OK = "OK";
 
+  /**
+   * Configuration keys for managing the scheduler
+   */
+  public static final String KEY_OF_SCHEDULER_RELEASE_TIMEOUT_MS = "pinot.query.scheduler.release.timeout.ms";
+  public static final long DEFAULT_SCHEDULER_RELEASE_TIMEOUT_MS = TimeUnit.MINUTES.toSeconds(1);
```

Comment:
could you comment on why this default value is being chosen. I am sensing it is related to the default timeout, but if someone set timeout = 120second. isn't this going to cause problem?

Reply:
The idea was to make sure this timeout is not hit in regular processing - I will set the default to `-1` and handle that appropriately. At some point, when we're ready to enable the defensive scheduling, we can give more thought to what we want this timeout to be.